central/reports/scheduler/v2/scheduler.go

```diff
@@ -0,0 +1,33 @@
+package v2
+
+import (
+	"context"
+
+	reportGen "github.com/stackrox/rox/central/reports/scheduler/v2/reportgenerator"
+	"github.com/stackrox/rox/generated/storage"
+)
+
+// Scheduler maintains the schedules for reports
+//
+//go:generate mockgen-wrapper
+type Scheduler interface {
```

Comment:
Can you add a function to the scheduler interface to identify if the job is schedulable?

Reply:
Schedulable meaning "can on demand report be submitted or not" ?

Reply:
That is correct. It should be the scheduler to determine whether a report is schedulable and not the datastore.

Reply:
Oh yes it is the scheduler that identifies whether a report can be submitted or not. There is a function `doesUserHavePendingReport` called from `SubmitReportRequest` implementation that checks if a report can be submitted or not. I will expose it via the Scheduler interface .